includes/class-wc-bfx-pay-gateway.php

```diff
@@ -501,42 +516,20 @@ public function webhook()
         $payload = file_get_contents('php://input');
         $data = json_decode($payload, true);
         $order = wc_get_order($data['orderId']);
-        if ('COMPLETED' !== $data['status']) {
-            $order->update_status('failed');
-
-            return;
-        }
-        ob_start();
-
-        $order->payment_complete();
 
         if ($this->debug) {
-            update_option('webhook_debug', $payload);
             $logger = wc_get_logger();
             $logger->info('WEBHOOK CALL >> '.wc_print_r($payload, true), ['source' => 'bitfinex-pay']);
         }
-        ob_clean();
-        status_header(200);
-        echo 'true';
-        exit();
-    }
 
-    /**
-     * Change payment complete order status to completed for Вitfinex payments method orders.
-     *
-     * @param string         $status   current order status
-     * @param int            $order_id order ID
-     * @param WC_Order|false $order    order object
-     *
-     * @return string
-     */
-    public function change_payment_complete_order_status($status, $order_id = 0, $order = false)
-    {
-        if ($order && 'bfx_payment' === $order->get_payment_method()) {
-            $status = 'completed';
+        $is_completed = $this->update_order_status($order, $data['status']);
+
+        if ($is_completed) {
```

Comment:
hmm bfx pay backend actually ignores the status, we just sent `200 OK, true`  as convention that we acknowledged the webhook call, I think we should always send this no matter if order is completed or not

Reply:
Updated the code @vigan-abd Please check again!